utils/ratelimits.py

```diff
@@ -47,55 +30,44 @@ def __setitem__(self, key, value):
             'expire_time': self.expire_time
         }
 
-    def set(self, key, value):
-        self.cache[key] = {
-            'data': value,
-            'timestamp': datetime.now(),
-            'expire_time': self.expire_time
-        }
+    def expires_on(self, item):
+        c = self.cache[item]
 
+        return c['timestamp'] + c['expire_time']
 
-cache = RatelimitCache()
+    def set(self, key, value):
+        return self.__setitem__(key, value)
 
 
-def get_config():
-    try:
-        with open('config.json') as config:
-            data = json.load(config)
-            if not isinstance(data, dict):
-                print('config.json must have a dict as base')
-                return {}
-            else:
-                return data
-    except FileNotFoundError:
-        print('config.json wasn\'t found in the current directory')
-        return {}
+cache = RatelimitCache()
 
 
 def ratelimit(func, max_usage=5):
     def wrapper(*args, **kwargs):
         key = request.headers.get('authorization', None)
-        try:
-            if get_config().get('memer_key') == key:
-                is_memer = True
-            else:
-                is_memer = False
-        except KeyError:
-            print('memer-key must exist in the config')
-            raise KeyError
-        if key in cache:
+        if key.endswith('-unlimited'):
+            unlimited = True
+        else:
+            unlimited = False
+        if unlimited:
+            return make_response((func(*args, **kwargs), 200, {"X-RateLimit-Limit": 'Unlimited',
+                                                               "X-RateLimit-Remaining": "Unlimited",
+                                                               "X-RateLimit-Reset": 2147483647}))
+        if key in cache and cache[key]:
```

Comment:
What's the need for `cache[key]`?

Reply:
For some reason, cache.get(key) will return NoneType sometimes and I haven't figured out why exactly. This bodge doesn't appear to cause issues and it works

Reply:
This may be related to `__getitem__` deleting a key if the key is expired